tests/integration/pilot/mcs/discoverability/discoverability_test.go

```diff
@@ -415,18 +382,23 @@ func createAndCleanupServiceExport(t framework.TestContext, service string, expo
 		}
 	} else {
 		scopes.Framework.Infof("No MCS Controller running. Manually creating ServiceImport in each cluster")
-		for _, c := range importClusters {
-			c := c
-			g.Go(func() error {
-				// Generate a dummy service in the cluster to reserve the ClusterSet VIP.
-				clusterSetIPSvc, err := genClusterSetIPService(c)
-				if err != nil {
-					return err
+		for _, c := range exportClusters {
+			svc, err := c.Kube().CoreV1().Services(echos.Namespace.Name()).Get(context.TODO(), common.ServiceB, metav1.GetOptions{})
+			if err != nil {
+				t.Errorf("failed to get %s/%s: %s", common.ServiceB, echos.Namespace.Name(), err)
+			}
+			// When there is no MCS controller, we are not able to create ServiceImport's VIP that would route traffic
+			// across services within the cluster set. Therefore, a ServiceEntry is created only in one of the clusters.
+			importClusters := exportClusters.Exclude(c)
+			for _, c := range importClusters {
+				// Make service B resolvable in the domain clusterset.local
+				if c.IsPrimary() {
+					createServiceEntry(t, c, common.ServiceB, echos.Namespace.Name())
 				}
-
-				// Create a ServiceImport in the cluster with the ClusterSet VIP.
-				return createServiceImport(c, clusterSetIPSvc.Spec.ClusterIP, serviceImportGVR)
-			})
+				if err := createServiceImport(c, svc.Spec.ClusterIP, serviceImportGVR); err != nil {
```

Comment:
Now ServiceImport has VIP set to cluster IP of an exported service.

Reply:
Doesn't that mean we cannot distinguish the two at runtime?

Reply:
Since I think for TCP envoy is matching based on IP?

Reply:
You can ignore this, because I wasn't right. I don't know why I was getting errors during manual tests, but actually current implementation works and most of my changes are not necessary. Please look at this PR: #44722.